File: merge_text.pl
#!/usr/bin/env perl

use File::Spec;
use File::Slurp;

@sections = (
    File::Spec->catfile("text", "supporting", "front.md"),
    File::Spec->catfile("text", "median", "001.md"),
    File::Spec->catfile("text", "median", "002.md"),
    File::Spec->catfile("text", "median", "003.md"),
);

open(IO, '>book.md');

my $i = 0;
my $n = $#sections;

foreach my $section (@sections) {
    $i = $i + 1;
    my $text = read_file($section);
    print IO $text;
    if ($i <= $n) {
         print IO "\n\n";
         print IO "---";
         print IO "\n\n";
    }
}

close(IO);

File: merge_text.jl
#!/usr/bin/env julia

sections = [
	joinpath("text", "supporting", "front.md"),
	joinpath("text", "median", "001.md"),
	joinpath("text", "median", "002.md"),
]

io = open("book.md", "w")

i = 0
n = length(sections)

for section in sections
	i += 1
	text = readall(section)
	print(io, text)
	if i != n
		print(io, "\n\n")
		print(io, "---")
		print(io, "\n\n")
	end
end

close(io)
